internal/pkg/policy/self.go

```diff
@@ -187,6 +204,89 @@ func (m *selfMonitorT) groupByLatest(policies []model.Policy) map[string]model.P
 	return groupByLatest(policies)
 }
 
+// sendPolicyOutput will parse the policy and send it through the config channel with only Output.Elasticsearch and RevisionIdx set
+// It will not send to the config channel if the policy revision_idx has not changed.
+// It returns any errors encountered when parsing the policy
+func (m *selfMonitorT) sendPolicyOutput() error {
+	// policy revision has not changed
+	if m.policy.RevisionIdx == m.lastRev {
+		return nil
+	}
+	// always copy revisionIdx
+	m.lastRev = m.policy.RevisionIdx
+
+	name, ok := getFleetOutputName(m.policy)
+	if !ok {
+		return fmt.Errorf("unable to find fleet-server use_output attribute")
+	}
+	data, ok := m.policy.Data.Outputs[name]
+	if !ok {
+		return fmt.Errorf("unable to find output name %q in policy", name)
+	}
+	outType, ok := data["type"].(string)
+	if !ok {
+		return fmt.Errorf("output name %s has non-string in type attribute: %w", name, ErrInvalidOutput)
+	}
+	if outType != OutputTypeElasticsearch {
+		return fmt.Errorf("output %s is type: %q, expected: elasticsearch", name, outType)
+	}
+
+	var policyES config.Elasticsearch
+	output, err := ucfg.NewFrom(data, config.DefaultOptions...)
+	if err != nil {
+		return fmt.Errorf("unable to create config from output data: %w", err)
+	}
+	if err := output.Unpack(&policyES, config.DefaultOptions...); err != nil {
+		return fmt.Errorf("unable to unback config data to config.Elasticsearch: %w", err)
+	}
+
+	// The output block in the policy may not have the schema set so we need to manually set it.
+	isHTTPS := false
+	for _, host := range policyES.Hosts {
+		if strings.HasPrefix(strings.ToLower(host), "https") {
+			isHTTPS = true
+			break
+		}
+	}
+	if isHTTPS {
+		policyES.Protocol = "https"
+	}
+	m.cfgCh <- &config.Config{
+		Output: config.Output{
+			Elasticsearch: policyES,
+		},
+		RevisionIdx: m.lastRev,
+	}
+	return nil
+}
+
+// getFleetOutputName returns the output name that the fleet-server input of the policy uses
+func getFleetOutputName(p *model.Policy) (string, bool) {
```

Comment:
@juliaElastic I've added this method so we use the `use_output` attribute of the fleet-server input to get the output name directly instead of scanning for the 1st `elasticsearch` type output